### ISSUE TITLE
bug is missing in props validation

### ISSUE THREAD

**Issue by djmisterjon** [yannickcr/eslint-plugin-react#3090]
hi why this is not valid ? it should ok no! 
![image](https://user-images.githubusercontent.com/24865815/135553248-9df26866-30a4-433c-bc75-1f50c9025149.png)
![image](https://user-images.githubusercontent.com/24865815/135553362-7c6e6474-5bc7-4020-8676-a223a610a00d.png)

here the demo from your example
```tsx
interface Props {
	age: number
 }
 const Hello: VFC<Props> = function Hello(props) {
	const { age } = props;

	return <div>Hello {name}</div>;
	// 'name' type is missing in props validation
 }

```

**Comment by ljharb** [comment]
Are you using the TypeScript eslint parser?

**Comment by djmisterjon** [comment]
yes
```js
{
    "env": {
        "browser": true,
        "es2021": true,
        "node": true
    },
    "extends": [
        "plugin:react/recommended",
        "plugin:react-hooks/recommended"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaFeatures": {
            "jsx": true
        },
        "ecmaVersion": 12,
        "sourceType": "module"
    },
    "plugins": [
        "react",
        "@typescript-eslint"
    ],
    "rules": {
        "array-bracket-spacing": "warn",
        "multiline-comment-style": "warn",
        "no-console": "warn",
        "comma-dangle": "off",
        "no-var": "warn",
        "no-duplicate-imports": "error",
        "sort-imports": "warn"
    }
}
```

```json
	"devDependencies": {
		"@types/chrome": "0.0.157",
		"@types/nw.js": "^0.13.14",
		"@types/react": "^17.0.20",
		"@types/react-dom": "^17.0.9",
		"@types/react-grid-layout": "^1.1.2",
		"@typescript-eslint/eslint-plugin": "^4.32.0",
		"@typescript-eslint/parser": "^4.32.0",
		"@vitejs/plugin-react-refresh": "^1.3.6",
		"commander": "^8.1.0",
		"eslint": "^7.32.0",
		"eslint-config-airbnb": "^18.2.1",
		"eslint-plugin-import": "^2.24.2",
		"eslint-plugin-jsx-a11y": "^6.4.1",
		"eslint-plugin-react": "^7.26.1",
		"eslint-plugin-react-hooks": "^4.2.0",
		"nw": "^0.55.0",
		"nw-builder": "^3.5.7",
		"ts-essentials": "^8.1.0",
		"vite": "^2.5.6"
	}
```

**Comment by ljharb** [comment]
That all seems good. Can you make a repro repo?

**Comment by djmisterjon** [comment]
> That all seems good. Can you make a repro repo?

oki i think i understand what happening.

```tsx
interface Props {
	age: number
 }
 const Hello: VoidFunctionComponent<Props> = function Hello(props) {
	const { age } = props;

	return <div>Hello {name}</div>;
	// 'name' type is missing in props validation
 }

 const Hello2: FunctionComponent<Props> = function Hello(props) {
	const { age } = props;

	return <div>Hello {name}</div>;
	// 'name' type is missing in props validation
 }
```

it seem `VoidFunctionComponent` are not supported by the rule ! ?
Work fine with `FunctionComponent` 

![image](https://user-images.githubusercontent.com/24865815/135639779-bd5bbe7d-8d89-4058-a28e-e3b21c8360bd.png)

**Comment by ljharb** [comment]
It may be an oversight. I think the plugin would need to be updated to handle it.
